src/agent/configuration_parser/src/configuration_parser.cpp

```diff
@@ -52,16 +24,21 @@ namespace configuration
     {
         try
         {
-            m_config = YAML::LoadFile(configFile.string());
+            m_config_file_path = configFile;
+            LoadLocalConfig(m_config_file_path);
+
+            signal_dispatcher::SignalDispatcher::GetInstance().RegisterListener("new_shared_configuration",
+                                                                                [this]() { ReloadConfiguration(); });
         }
         catch (const std::exception& e)
         {
+            m_config = YAML::Node();
             LogWarn("Using default values due to error parsing wazuh-agent.yml file: {}", e.what());
         }
     }
 
     ConfigurationParser::ConfigurationParser()
-        : ConfigurationParser(std::filesystem::path(CONFIG_FILE_NAME))
+        : ConfigurationParser(CONFIG_FILE)
     {
     }
 
```

Comment:
Is the function below being used anywhere?

Reply:
Constructor `ConfigurationParser()` and constructor `ConfigurationParser(const std::filesystem::path& configFilePath)` are used in the [Agent ](https://github.com/wazuh/wazuh-agent/blob/enhancement/263-agent-centralized-configuration-validate-and-apply-downloaded-files/src/agent/src/agent.cpp#L14) constructor, while constructor `ConfigurationParser(const std::string& stringToParse)` is currently used only for testing, when tests are refactored by simulating calls to the Yaml-cpp library this constructor will not be needed.